### ISSUE TITLE
boost failing in setup and cmake

### ISSUE THREAD

**Issue by pavlis** [wangyinz/mspass#200]
Running into a strange problem I don't know how to fix.   It happens in any attempt on this (ubuntu) machine to build mspass from source code.  i.e. cmake files and setup.py fail the same way.  

The problem is something about the configuration for autodownloading of boost.  When I run cmake I get this:
```
CMake Error at boost-prefix/src/boost-stamp/download-boost.cmake:170 (message):
  Each download failed!

    error: downloading 'https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz' failed
          status_code: 22
          status_string: "HTTP response code said error"
          log:
          --- LOG BEGIN ---
            Trying 34.223.179.172:443...
```
I get a similar error with setup.py, but this is more instructive:
```
(base) pavlis@pavlis-linux:~/src/mspass/cxx/build$ wget https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz
--2021-05-13 09:37:41--  https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz
Resolving dl.bintray.com (dl.bintray.com)... 52.43.14.61, 34.217.158.44, 52.42.217.145, ...
Connecting to dl.bintray.com (dl.bintray.com)|52.43.14.61|:443... connected.
HTTP request sent, awaiting response... 403 Forbidden
2021-05-13 09:37:41 ERROR 403: Forbidden.
```
For whatever reason the link to the dl.bintray.com does not resolve.   There is a valid download [at this url](https://boostorg.jfrog.io/artifactory/main/release/1.71.0/source/).   The solution is clearly in knowing how cmake is getting the configuration for where to download boost.   I'm hoping you, Ian, can fix this quickly since you set up the cmake configuration originally.

**Comment by pavlis** [comment]
Well, after I finished above I remember some about how this works and I found the issue in the file mspass/cxx/cmake/boost-download.cmake.   Here it is:
```
cmake_minimum_required(VERSION 3.4)
  
project(boost-download NONE)

include(ExternalProject)

ExternalProject_Add(
  boost
  SOURCE_DIR "@BOOST_DOWNLOAD_ROOT@/boost-src"
  URL
    https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz
  URL_HASH
    SHA256=96b34f7468f26a141f6020efb813f1a2f3dfb9797ecf76a7d7cbd843cc95f5bd
  UPDATE_COMMAND wget https://svn.boost.org/trac10/raw-attachment/ticket/11120/python_jam.patch && patch tools/build/src/tools/python.jam python_jam.patch
  CONFIGURE_COMMAND ./bootstrap.sh --prefix=${PROJECT_BINARY_DIR}
  BUILD_COMMAND ./b2 cxxflags=-fPIC --with-serialization -j 8
  BUILD_IN_SOURCE 1
  INSTALL_COMMAND ./b2 install
  TEST_COMMAND ""
  )
```
the URL variable is pointing at the url that seems to not be functioning.   I'd change this but you, Ian, may have a better idea what is the most stable download url.   I note, by the way, that boost is now at a 1.76 as of a month ago.  May want to update to the newer version if we have to fix this anyway.

**Comment by wangyinz** [comment]
Yes, that's exactly where the url is set. I don't know what is the most stable url, but the link you provide looks reasonable.

**Comment by pavlis** [comment]
Ok, I made that change with my local copy and it seems to work.  It might be easier if you made the change directly to master.  Here is what I changed:
```
 URL
    https://boostorg.jfrog.io/artifactory/main/release/1.71.0/source/boost_1_71_0.tar.gz
  URL_HASH
    SHA256=96b34f7468f26a141f6020efb813f1a2f3dfb9797ecf76a7d7cbd843cc95f5bd
```
URL was easier.  It was more of a challenge to find the HASH magic string.  It was in the json file  - why I don't know.

If making this to the master you might consider upgrading to 1.76.   Haven't looked at the change log to know if that would matter though.  Likely would not matter for existing code base.
